lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/ReadOnlySystemAccessControl.java

```diff
@@ -105,6 +110,17 @@ public void checkCanGrantExecuteFunctionPrivilege(SystemSecurityContext context,
     {
     }
 
+    @Override
+    public void checkCanGrantExecuteFunctionPrivilege(SystemSecurityContext context, FunctionKind functionKind, CatalogSchemaRoutineName functionName, TrinoPrincipal grantee, boolean grantOption)
+    {
+        if (functionKind==FunctionKind.TABLE) {
+            // May not be read-only, so deny
+            String granteeAsString = format("%s '%s'", grantee.getType().name().toLowerCase(Locale.ENGLISH), grantee.getName());
+            denyGrantExecuteFunctionPrivilege(functionName.toString(), context.getIdentity(), granteeAsString);
+        }
+        throw new UnsupportedOperationException();
```

Comment:
why throw exception needed?

Reply:
What other function kinds are being passed here? 
is it safe to assume they all are read-only?

Reply:
Making this future proof

```java
switch (functionKind) {
    case SCALAR:
    case AGGREGATE:
    case WINDOW:
        return;
    case TABLE:
        // May not be read-only, so deny
        String granteeAsString = format("%s '%s'", grantee.getType().name().toLowerCase(Locale.ENGLISH), grantee.getName());
        denyGrantExecuteFunctionPrivilege(functionName.toString(), context.getIdentity(), granteeAsString);
}
throw new UnsupportedOperationException("Unsupported function kind: " + functionKind);
```